ManagedScheduledExecutorService.schedule with Trigger always returning null cannot be canceled

When calling a ManagedScheduledExecutorService with a trigger that always returns null, calling cancel on such a future leads to NullPointerException:

Caused by: java.lang.NullPointerException
at org.glassfish.enterprise.concurrent.internal.ManagedScheduledThreadPoolExecutor$TriggerControllerFuture.cancel(ManagedScheduledThreadPoolExecutor.java:662) [javax.enterprise.concurrent-1.0.jar:]
...
#### Environment
Wildfly 9.0.1

--- a/src/main/java/org/glassfish/enterprise/concurrent/internal/ManagedScheduledThreadPoolExecutor.java
+++ b/src/main/java/org/glassfish/enterprise/concurrent/internal/ManagedScheduledThreadPoolExecutor.java
@@ -635,8 +635,12 @@ public class ManagedScheduledThreadPoolExecutor extends ScheduledThreadPoolExecu
         @Override
         public boolean cancel(boolean mayInterruptIfRunning) {
             super.cancel(mayInterruptIfRunning);
-            // cancel the next scheduled task
-            return getCurrentFuture().cancel(mayInterruptIfRunning);
+            // cancel the next scheduled task if there is one
+            ManagedTriggerSingleFutureTask<V> future = getCurrentFuture();
+            if (future != null) {
+                return future.cancel(mayInterruptIfRunning);
+            }
+            return true;
         }
 
         @Override

